packages/search-ui-app-search-connector/src/AppSearchAPIConnector.ts

```diff
@@ -3,6 +3,39 @@ import * as ElasticAppSearch from "@elastic/app-search-javascript";
 import { adaptResponse } from "./responseAdapter";
 import { adaptRequest } from "./requestAdapters";
 import buildResponseAdapterOptions from "./buildResponseAdapterOptions";
+import type {
+  QueryConfig,
+  RequestState,
+  SearchState,
+  AutocompleteQuery,
+  SuggestionsQueryConfig
+} from "@elastic/search-ui";
+
+export type AppSearchAPIConnectorParams = {
+  searchKey: string;
+  engineName: string;
+  hostIdentifier: string;
+  beforeSearchCall?: SearchQueryHook;
+  beforeAutocompleteResultsCall?: SearchQueryHook;
+  beforeAutocompleteSuggestionsCall?: SuggestionsQueryHook;
+  endpointBase?: string;
+};
+
+interface ResultClickParams {
+  query: string;
+  documentId: string;
+  requestId: string;
+  tags: string[];
+}
+
+export type SearchQueryHook = (
+  queryOptions: QueryConfig,
+  next: (newQueryOptions: any) => any
+) => any;
+export type SuggestionsQueryHook = (
```

Comment:
is the return type here void?

Reply:
Nice catch! I updated return type of hooks in this commit: https://github.com/elastic/search-ui/pull/628/commits/2e24a3f4b58d2abfd3a95f0d62a6701b8d5a68df

I had to still keep `any` for the return type of `next`, because we're returning an adapted response there:

```
return this.beforeSearchCall(options, async (newOptions) => {
      const response = await this.client.search(query, newOptions);
      return adaptResponse(response, buildResponseAdapterOptions(queryConfig));
    });
```

I don't know the exact type since it's partially coming from a javascript client that doesn't have types yet, thus I have to keep `any`.

Reply:
Nice catch! I went back and forth on this a bit (you can see in the last commits) but decided to keep `any` everywhere. Some `void` types are causing errors in other places and I'm not confident in assigning more specific types. Keeping it as-is for now.